Formula/p/purr.rb

```diff
@@ -0,0 +1,26 @@
+class Purr < Formula
+  desc "Versatile zsh CLI tool for viewing and searching through Android logcat output"
+  homepage "https://github.com/google/purr"
+  url "https://github.com/google/purr/archive/refs/tags/2.0.0.tar.gz"
+  sha256 "f4e6c3dcdd386da7d3664fe6c947e9cfdb2e3362ca8c78e47d0445d94d471246"
+  license "Apache-2.0"
+  head "https://github.com/google/purr.git", branch: "main"
+
+  depends_on "make" => :build
+  depends_on "fzf"
+  depends_on "zsh"
+
+  def install
+    system "make", "purr"
+    bin.install "out/purr"
+  end
+
+  def caveats
+    <<~EOS
+      purr requires you to have adb executable in $PATH
+
+      adb is part of Android SDK command line tools. Get them at:
+        https://developer.android.com/studio#command-line-tools-only
+    EOS
+  end
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.